language/sygus.tex

```diff
@@ -1898,13 +1898,20 @@ \subsection{Satisfying Semantic Specifications}
 
 \section{Calling Oracles}
 \label{sec:oracleimplementations}
-TODO: information on how to call oracles. Are these binaries. input format? 
 
-We say that a binary with name $N$ \emph{implements a function of sort
-$\sigma_1 \times \ldots \times \sigma_n \rightarrow 
-\sigma_{n+1} \times \ldots \times \sigma_{n+m}$}
-if
-$\ldots$ TODO $\ldots$
+The SyGuS solver makes queries to an oracle via a text file with the extension \textit{.oracle}. The oracle is called by the solver with the command \textit{oracle-name textfile.oracle}.
```

Comment:
I think we need to keep the definition "implements a function of sort ..." here, since this definition is referred to in the description of the commands.  
Also, it would be helpful to enclose this paragraph in a \begin{def} / \end{def} block.

Reply:
Sorry for the slow response. I've put the implements an interface back in (as discussed by email, we don't require it to be functional). 

There isn't a definition environment in this latex doc, is there?

Reply:
I'm not sure, this tex is actually copied from the original v1 document :)
If not, can you introduce it?

Reply:
Hmm I did it, but it looks weird. Why is this the only definition in the document? This is the diff, is this what you want? 

~~~
diff --git a/language/macros.tex b/language/macros.tex
index f9f058c..052034c 100644
--- a/language/macros.tex
+++ b/language/macros.tex
@@ -2,6 +2,7 @@
 \newtheorem{example}{Example}
 \newcommand{\rem}[1]{\textcolor{red}{[#1]}}
 \newcommand{\ajr}[1]{\rem{#1 --ajr}}
+\newtheorem{definition}{Definition}
 
 \global\long\def\autobox#1{#1}
 
diff --git a/language/sygus.tex b/language/sygus.tex
index 57a7b96..11a8b79 100644
--- a/language/sygus.tex
+++ b/language/sygus.tex
@@ -1899,7 +1899,9 @@ involve (defined) macros and recursive functions.
 \section{Calling Oracles}
 \label{sec:oracleimplementations}
 
+\begin{definition}
 The oracle is a binary that implements an iterface of sort $\sigma_1 \times \ldots \times \sigma_n \rightarrow \sigma$.
+\end{definition}
 The SyGuS solver ma
~~~

Reply:
Yes, that looks right.  Why does it look weird?

Reply:
Ok, pushed. I just felt weird about it being the only definition in the doc. Surely this isn't the only thing that needed defining?

Reply:
Yes, I have a similar feeling. 

Regardless having the definition should make things easier to clarify, and perhaps other parts of the document could be made more precise in the same way.